cmd/snap-device-helper/snap-device-helper-test.c

```diff
@@ -282,6 +282,94 @@ static void test_sdh_action_nvme(sdh_test_fixture *fixture, gconstpointer test_d
     }
 }
 
+static void test_sdh_action_remove_fallback_devtype(sdh_test_fixture *fixture, gconstpointer test_data) {
+    /* check that fallback guessing of device type if applied during remove action */
+    mkdir_in_sysroot(fixture, "/sys/devices/pci0000:00/0000:00:01.1/0000:01:00.0/nvme/nvme0/nvme0n1");
+    mkdir_in_sysroot(fixture, "/sys/devices/pci0000:00/0000:00:01.1/0000:01:00.0/nvme/nvme0/nvme0n1p1");
+    mkdir_in_sysroot(fixture, "/sys/devices/pci0000:00/0000:00:01.1/0000:01:00.0/nvme/nvme0/ng0n1");
+    mkdir_in_sysroot(fixture, "/sys/devices/pci0000:00/0000:00:01.1/0000:01:00.0/nvme/nvme0/hwmon0");
+    mkdir_in_sysroot(fixture, "/sys/devices/foo/block/sda/sda4");
+    mkdir_in_sysroot(fixture, "/sys//devices/pnp0/00:04/tty/ttyS0");
+
```

Comment:
I've got some `nvmem` entries in my `/sys/` directories that aren't related to nvme devices at all:

```
$ sudo find /sys -name '*nvmem*' -ls
    25642      0 -rw-r--r--   1 root     root            0 Dec 16 20:15 /sys/devices/pnp0/00:04/cmos_nvram0/nvmem
    11935      0 drwxr-xr-x   4 root     root            0 Dec 16 20:43 /sys/bus/nvmem
```

This is just a laptop, nothing as exotic as Optane DC Persistent Memory dimms on board. I don't think any of this current code would make a mistake around the `nvme` vs `nvmem` difference, but it might be nice to encode some unrelated nvmem devices in the test case to make sure it stays that way.

Reply:
AFAICT this shoudl appear on every PC-like hardware, but it's only used by rtc and there even isn't an allocated device major/minor for this even